integration-tests-ex/cases/src/test/java/org/apache/druid/testsEx/msq/ITS3SQLBasedIngestionTest.java

```diff
@@ -0,0 +1,206 @@
+        .withRegion(System.getenv("AWS_REGION"))
+        .build();
+  }
+
+  public static String[] fileList()
+  {
+    return new String[] {
+        WIKIPEDIA_DATA_1, WIKIPEDIA_DATA_2, WIKIPEDIA_DATA_3
+    };
+  }
+
+  @BeforeClass
+  public static void uploadDataFilesToS3()
+  {
+    String localPath = "resources/data/batch_index/json/";
+    for (String file : fileList()) {
+      s3Client.putObject(
+          System.getenv("DRUID_CLOUD_BUCKET"),
+          System.getenv("DRUID_CLOUD_PATH") + "/" + file,
+          new File(localPath + file)
+      );
+    }
+  }
+
+  @AfterClass
+  public static void deleteFilesFromS3()
+  {
+    // Delete uploaded data files
+    DeleteObjectsRequest delObjReq = new DeleteObjectsRequest(System.getenv("DRUID_CLOUD_BUCKET"))
+        .withKeys(fileList());
+    s3Client.deleteObjects(delObjReq);
+
+    // Delete segments created by druid
+    ListObjectsRequest listObjectsRequest = new ListObjectsRequest()
+        .withBucketName(System.getenv("DRUID_CLOUD_BUCKET"))
+        .withPrefix(System.getenv("DRUID_CLOUD_PATH") + "/" + datasource + "/");
+
+    ObjectListing objectListing = s3Client.listObjects(listObjectsRequest);
+
+    while (true) {
+      for (S3ObjectSummary objectSummary : objectListing.getObjectSummaries()) {
+        s3Client.deleteObject(System.getenv("DRUID_CLOUD_BUCKET"), objectSummary.getKey());
+      }
+      if (objectListing.isTruncated()) {
+        objectListing = s3Client.listNextBatchOfObjects(objectListing);
+      } else {
+        break;
+      }
+    }
+  }
+
+  @Test
+  @Parameters(method = "test_cases")
+  @TestCaseName("Test_{index} ({0})")
+  public void testSQLBasedBatchIngestion(Pair<String, List> s3InputSource)
+  {
+    try {
+      String sqlTask = getStringFromFileAndReplaceDatasource(CLOUD_INGEST_SQL, datasource);
+      String inputSourceValue = jsonMapper.writeValueAsString(s3InputSource.rhs);
+      Map<String, Object> context = ImmutableMap.of("finalizeAggregations", false,
+                                                    "maxNumTasks", 5,
+                                                    "groupByEnableMultiValueUnnesting", false);
+
+      sqlTask = StringUtils.replace(
+          sqlTask,
+          "%%INPUT_SOURCE_PROPERTY_KEY%%",
+          s3InputSource.lhs
+      );
+      sqlTask = StringUtils.replace(
+          sqlTask,
+          "%%INPUT_SOURCE_PROPERTY_VALUE%%",
+          inputSourceValue
+      );
+
+      // Setting the correct object path in the sqlTask.
+      sqlTask = StringUtils.replace(
+          sqlTask,
+          "%%BUCKET%%",
+          config.getCloudBucket() // Getting from DRUID_CLOUD_BUCKET env variable
+      );
+      sqlTask = StringUtils.replace(
+          sqlTask,
+          "%%PATH%%",
+          config.getCloudPath() // Getting from DRUID_CLOUD_PATH env variable
+      );
+
+      submitTask(sqlTask, datasource, context);
+      doTestQuery(INDEX_QUERIES_FILE, datasource);
```

Comment:
Does this verify the results by running queries against the uploaded data? If so, please add a comment to note this.

If this does not verify results, then we need to add a step to do so. Else, this is not much of a test since a simple direct return from `submitTask()` (while doing nothing) would count as a success.